### PR TITLE
Upload a bundle via action bar. 

### PR DESCRIPTION
**Pull request by Raisins**
Be able to upload a bundle. Current flow go into action bar and type `upload` then `file` press command/cnt+enter a modal will appear allowing you to select a file. 

- [x] Modal with upload form
- [x] way to access modal via action bar. (note this will change with new action bar)
- [ ] passing info to the cli
- [x] updating user to progress
- [x] update worksheet on success. 

issue #1030

### CODE REVIEW

File: codalab/apps/api/views/worksheet_views.py [hunk 1]
```diff
@@ -435,27 +435,40 @@ def post(self, request):
...
-            tb = traceback.format_exc()
-            logging.error(tb)
-            logging.debug('-------------------------')
-            return Response({"error": smart_str(e)}, status=500)
+        if request.FILES:
+            try:
+                service.upload_bundle_file(request.FILES['file'], request.POST['worksheet_uuid'])
+                return Response({'success': True}, content_type="application/json")
+            except Exception, e:
+                logging.error(self.__str__())
```
Comment (ckcollab) [inline codalab/apps/api/views/worksheet_views.py]: I see this repeated a lot, can we make this into some kind of utility function?

File: codalab/apps/api/views/worksheet_views.py [hunk 2]
```diff
@@ -435,27 +435,40 @@ def post(self, request):
 
 class BundleUploadApi(views.APIView):
```
Comment (ckcollab) [inline codalab/apps/api/views/worksheet_views.py]: What is the file size limit? We get questions on this all the time for competition bundles, submissions, etc. -- do we know for sure what it is here?

File: codalab/apps/web/bundles.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 from django.conf import settings
 from django.template.defaultfilters import slugify
+from django.utils.encoding import smart_str
```
Comment (ckcollab) [inline codalab/apps/web/bundles.py]: woo!

File: codalab/apps/web/bundles.py [hunk 4]
```diff
@@ -156,6 +159,28 @@ def upload_bundle_url(self, url, info, worksheet_uuid):
...
+                'bundle_type': 'dataset',
+                'metadata': {
+                    'source_url': u'',
+                    'tags': [],
+                    'name': smart_str(file.name),
+                    'license': u'',
+                    'description': 'Upload from web'
+                }
+            }
+            #TODO save to worksheet.
```
Comment (Raisins) [inline codalab/apps/web/bundles.py]: @percyliang

**Pull request closed without merging**